spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ReactiveKafkaAutoConfiguration.java

```diff
@@ -0,0 +1,128 @@
+
+	private final KafkaProperties kafkaProperties;
+
+	private final ReactiveKafkaProperties reactiveKafkaProperties;
+
+	public ReactiveKafkaAutoConfiguration(KafkaProperties kafkaProperties,
+			ReactiveKafkaProperties reactiveKafkaProperties) {
+		this.kafkaProperties = kafkaProperties;
+		this.reactiveKafkaProperties = reactiveKafkaProperties;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(ReceiverOptions.class)
+	public <K, V> ReceiverOptions<K, V> receiverOptions() {
+		Map<String, Object> properties = this.kafkaProperties.buildConsumerProperties();
+		properties.putAll(this.reactiveKafkaProperties.buildReceiverProperties());
+		ReceiverOptions<K, V> receiverOptions = ReceiverOptions.create(properties);
+		int atmostOnceCommitAheadSize = this.reactiveKafkaProperties.getReceiver().getAtmostOnceCommitAheadSize();
+		if (atmostOnceCommitAheadSize >= 0) {
+			receiverOptions = receiverOptions.atmostOnceCommitAheadSize(atmostOnceCommitAheadSize);
+		}
+		Optional<Duration> closeTimeout = Optional
+				.ofNullable(this.reactiveKafkaProperties.getReceiver().getCloseTimeout());
+		if (closeTimeout.isPresent()) {
+			receiverOptions = receiverOptions.closeTimeout(closeTimeout.get());
+		}
+		Optional<Duration> pollTimeout = Optional
+				.ofNullable(this.reactiveKafkaProperties.getReceiver().getPollTimeout());
+		if (pollTimeout.isPresent()) {
+			receiverOptions = receiverOptions.pollTimeout(pollTimeout.get());
+		}
+		int maxDeferredCommits = this.reactiveKafkaProperties.getReceiver().getMaxDeferredCommits();
+		if (maxDeferredCommits >= 0) {
+			receiverOptions = receiverOptions.maxDeferredCommits(maxDeferredCommits);
+		}
+		int maxCommitAttempts = this.reactiveKafkaProperties.getReceiver().getMaxCommitAttempts();
+		if (maxCommitAttempts >= 0) {
+			receiverOptions = receiverOptions.maxCommitAttempts(maxCommitAttempts);
+		}
+		int commitBatchSize = this.reactiveKafkaProperties.getReceiver().getCommitBatchSize();
+		if (commitBatchSize >= 0) {
+			receiverOptions = receiverOptions.commitBatchSize(commitBatchSize);
+		}
+		Duration commitInterval = this.reactiveKafkaProperties.getReceiver().getCommitInterval();
+		if (commitInterval != null) {
+			receiverOptions = receiverOptions.commitInterval(commitInterval);
+		}
+		Collection<String> subscribeTopics = this.reactiveKafkaProperties.getReceiver().getSubscribeTopics();
+		if (subscribeTopics != null) {
+			receiverOptions = receiverOptions.subscription(subscribeTopics);
+		}
+		else {
+			Pattern subscribePattern = this.reactiveKafkaProperties.getReceiver().getSubscribePattern();
+			if (subscribePattern != null) {
+				receiverOptions = receiverOptions.subscription(subscribePattern);
+			}
+		}
+		return receiverOptions;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(SenderOptions.class)
+	public <K, V> SenderOptions<K, V> senderOptions() {
+		Map<String, Object> properties = this.kafkaProperties.buildProducerProperties();
+		properties.putAll(this.reactiveKafkaProperties.buildSenderProperties());
```

Comment:
Ditto - we shouldn't be adding these; they are reactor-kafka specific properties.